sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/aggregate/HyperLogLogPlusPlusSuite.scala

```diff
@@ -175,4 +195,45 @@ class HyperLogLogPlusPlusSuite extends SparkFunSuite {
     hll.update(buffer, input)
     evaluateEstimate(hll, buffer, 1);
   }
+
+  test("SPARK-16484: Test sketch evaluation") {
+    val (hll, input, buffer) = createSketchEstimator(0.05, DoubleType)
+    List(1, 1, 2, 2, 3).foreach(i => {
+      input.setDouble(0, i)
+      hll.update(buffer, input)
+    })
+    val sketch = hll.eval(buffer).asInstanceOf[Array[Byte]]
+    val bb = ByteBuffer.wrap(sketch)
+    assert(bb.getInt() == 1)
+    assert(bb.getDouble() == 0.05)
```

Comment:
same, this performs floating-point comparison; should we use an epsilon for determinism in the testing?

Reply:
I left this unmodified as I think comparisons of that 0.05 value are deterministic, and I don't expect the default relativeSD value to change.